nano/node/json_handler.cpp

```diff
@@ -4785,7 +4748,32 @@ void nano::json_handler::work_generate ()
 		{
 			ec = nano::error_rpc::difficulty_limit;
 		}
-		if (!ec)
+		// Retrieving optional block
+		std::shared_ptr<nano::block> block;
+		boost::optional<std::string> block_text (request.get_optional<std::string> ("block"));
+		if (!ec && block_text.is_initialized ())
+		{
+			block = block_impl (true);
+			if (block != nullptr)
+			{
+				if (hash != block->root ())
+				{
+					ec = nano::error_rpc::block_root_mismatch;
+				}
+				if (!ec && block->difficulty () >= difficulty)
+				{
+					// If optional block difficulty if higher than requested difficulty, send response immediately
```

Comment:
Small typo, should read "**is** higher than"

Reply:
Updated